qubes/tests/__init__.py

```diff
@@ -1328,33 +1328,54 @@ def prepare_hvm_system_linux(self, vm, init_script, extra_files=None):
 
     def create_bootable_iso(self):
         """Create simple bootable ISO image.
-        Type 'poweroff' to it to terminate that VM.
+        Type 'halt' to it to terminate that VM.
         """
-        isolinux_cfg = (
-            'prompt 1\n'
-            'label poweroff\n'
-            '   kernel poweroff.c32\n'
-        )
         output_fd, output_path = tempfile.mkstemp('.iso')
         with tempfile.TemporaryDirectory() as tmp_dir:
+            f = os.open(tmp_dir.name)
             try:
-                shutil.copy('/usr/share/syslinux/isolinux.bin', tmp_dir)
-                shutil.copy('/usr/share/syslinux/ldlinux.c32', tmp_dir)
-                shutil.copy('/usr/share/syslinux/poweroff.c32', tmp_dir)
-                with open(os.path.join(tmp_dir, 'isolinux.cfg'), 'w') as cfg:
-                    cfg.write(isolinux_cfg)
-                subprocess.check_call(['genisoimage', '-o', output_path,
-                                       '-c', 'boot.cat',
-                                       '-b', 'isolinux.bin',
+                os.mkdir('os', mode=0o755, dir_fd=f)
+                os.mkdir('iso', mode=0o755, dir_fd=f)
+                os.mkdir('os/images', mode=0o755, dir_fd=f)
+                os.mkdir('os/EFI', mode=0o755, dir_fd=f)
+                os.mkdir('os/EFI/BOOT', mode=0o755, dir_fd=f)
+                subprocess.check_call(['sudo', 'grub2-mkimage',
```

Comment:
Does it really need sudo? It shouldn't be needed if you write to the tmp_dir already. Which BTW is also missing here - according to the docs, default is stdout, and it isn't redirected here...